src/main/java/com/microsoft/cse/helium/app/utils/ParameterValidator.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.cse.helium.app.utils;
+
+import java.util.regex.Pattern;
+import org.springframework.stereotype.Component;
+
+/** ParameterValidator. */
+@Component
+public class ParameterValidator {
+
+  /* Valid input: starts with 'nm' (case sensitive)
+      followed by 5-9 digits
+      parses to int > 0
+  */
+  private static final String validActorRegex = "[nm]{2}[0-9]{5,9}";
+
+  /** isValidActorId. */
+  public Boolean isValidActorId(String actorId) {
+    Pattern p = Pattern.compile(validActorRegex);
+    return p.matcher(actorId).matches();
+  }
+
+  /** isValidSearchQuery. */
+  public Boolean isValidSearchQuery(String query) {
+    Boolean isValid = true;
```

Comment:
The above code can be simplified a bit. Whenever we do parameter validation, we should aim to return immediately on hitting an error. Essentially, stop on fail, so the following is pretty standard way of writing such code.
isValidSearchQuery(String query) {
    if(...) {
        return false 
   }
  return true
}

Reply:
This is done to maintain consistency with other validation methods where true or false is calculated from multiple decision trees.